src/models/subgraph-entities/vX/project.ts

```diff
@@ -68,122 +39,79 @@ type BaseProject = {
-    | 'veNftContract'
-  > & {
-    participants: ParticipantJson[]
-    printPremineEvents: MintTokensEventJson[]
-    payEvents: PayEventJson[]
-    tapEvents: TapEventJson[]
-    redeemEvents: RedeemEventJson[]
-    printReservesEvents: PrintReservesEventJson[]
-    deployedERC20Events: DeployedERC20EventJson[]
-    distributeToPayoutModEvents: DistributeToPayoutModEventJson[]
-    distributeToTicketModEvents: DistributeToTicketModEventJson[]
-    v1ConfigureEvents: V1ConfigureEventJson[]
-    configureEvents: ConfigureEventJson[]
-    veNftContract: VeNftContractJson
-    trendingScore: string
-    trendingVolume: string
-    totalPaid: string
-    totalPaidUSD: string
-    totalRedeemed: string
-    totalRedeemedUSD: string
-    currentBalance: string
-  }
->
-
-export const parseProjectJson = (project: ProjectJson): Partial<Project> =>
-  ({
-    ...project,
-    currentBalance: project.currentBalance
-      ? BigNumber.from(project.currentBalance)
-      : undefined,
-    totalPaid: project.totalPaid
-      ? BigNumber.from(project.totalPaid)
-      : undefined,
-    totalPaidUSD: project.totalPaidUSD
-      ? BigNumber.from(project.totalPaidUSD)
-      : undefined,
-    totalRedeemed: project.totalRedeemed
-      ? BigNumber.from(project.totalRedeemed)
-      : undefined,
-    totalRedeemedUSD: project.totalRedeemedUSD
-      ? BigNumber.from(project.totalRedeemedUSD)
-      : undefined,
-    participants: project.participants?.map(parseParticipantJson) ?? undefined,
-    printPremineEvents:
-      project.printPremineEvents?.map(parseMintTokensEventJson) ?? undefined,
-    payEvents: project.payEvents?.map(parsePayEventJson) ?? undefined,
-    tapEvents: project.tapEvents?.map(parseTapEventJson) ?? undefined,
-    redeemEvents: project.redeemEvents?.map(parseRedeemEventJson) ?? undefined,
-    configureEvents:
-      project.configureEvents?.map(parseConfigureEventJson) ?? undefined,
-    v1ConfigureEvents:
-      project.v1ConfigureEvents?.map(parseV1ConfigureEventJson) ?? undefined,
-    printReservesEvents:
-      project.printReservesEvents?.map(parsePrintReservesEventJson) ??
-      undefined,
-    deployedERC20Events:
-      project.deployedERC20Events?.map(parseDeployedERC20EventJson) ??
-      undefined,
-    distributeToPayoutModEvents:
-      project.distributeToPayoutModEvents?.map(
-        parseDistributeToPayoutModEvent,
-      ) ?? undefined,
-    distributeToTicketModEvents:
-      project.distributeToTicketModEvents?.map(
-        parseDistributeToTicketModEvent,
-      ) ?? undefined,
-    veNftContract: project.veNftContract
-      ? parseVeNftContractJson(project.veNftContract)
-      : undefined,
-    trendingVolume: project.trendingVolume
-      ? BigNumber.from(project.trendingVolume)
-      : undefined,
-    trendingScore: project.trendingScore
-      ? BigNumber.from(project.trendingScore)
-      : undefined,
-  } as Project)
+export const parseProjectJson = (j: Json<Project>): Partial<Project> => ({
```

Comment:
typing is great in long files like this. commenting out any one of the lines below should produce a type error